### PR TITLE
Jetpack Setup Wizard: Add Income Question Page

### PR DESCRIPTION
**Pull request by robertf4**
#### Changes proposed in this Pull Request:

This PR builds on the work in https://github.com/Automattic/jetpack/pull/15429 by adding the income question page.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Project: p1HpG7-8BF-p2
Design: pbtFPC-ol-p2

Figma design:
image

Current state:
image

Figma design mobile:
image

Current state mobile:
image

#### Testing instructions:
1. Add the following to the end of wp-config.php: add_filter( 'jetpack_show_setup_wizard', '__return_true' );
2. Visit the Jetpack plugin and visit the `/setup` page.
3. Click either "Personal Use" or "Business Use". Verify that you advance to the `/setup/income` page.
4. Verify that back/forward works between these two pages.
5. Verify that you can select various choices both with the mouse and by tabbing with the keyboard.
6. Shrink the window to <= 660px. Verify that the page adjusts according to the design.
7. Verify that you can expand and hide the details.
8. Verify that you can select various choices with both the mouse and the keyboard.

#### Proposed changelog entry for your changes:
* Added new setup wizard

### CODE REVIEW

File: _inc/client/setup-wizard/checklist-answer/style.scss
Comment (robertf4) [reply _inc/client/setup-wizard/checklist-answer/style.scss:2]: Sure, I changed several of the colors to use declarations from that file.

**Comment by robertf4** [comment]
I'm going to rebase this PR against https://github.com/Automattic/jetpack/pull/15514 to keep things moving along. Once that PR merges into master I will rebase against master.

File: _inc/client/setup-wizard/income-question/index.jsx
Comment (jeherve) [inline _inc/client/setup-wizard/income-question/index.jsx:23]: I get a broken image here; removing the `/` seems to fix this.

```suggestion
				src={ imagePath + 'generating-cash.svg' }
```

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by jeffgolenski** [comment]
@robertf4 The functionality here is beautiful. As mentioned in slack + my walkthrough video.

For this page specifically, I think we just need to tighten up a couple of small items:

* Assign a `max-width` of 650—700px + `margin: 0 auto;` to the container for all of the checkbox areas
* Vertically align the label text to the checkboxes (bump it up just a couple of pixels)

**Comment by robertf4** [comment]
Rebased against master.

File: _inc/client/setup-wizard/checklist-answer/index.jsx
Comment (jeherve) [inline _inc/client/setup-wizard/checklist-answer/index.jsx:15]: As we move away from classes in other areas of Jetpack, should we do the same here as well? 

Related: #15694.

File: _inc/client/setup-wizard/intro-page/index.jsx
Comment (jeherve) [inline _inc/client/setup-wizard/intro-page/index.jsx:41]: We'll most likely want to add tracking to those links so we can see how people navigate through the wizard.

**Review by jeherve — CHANGES_REQUESTED** [review]
This tests well for me, although it seems like some padding is missing above each checkbox:

image

I'll let Jeff weigh in on that though, I'm no designer.

File: _inc/client/setup-wizard/income-question/index.jsx
Comment (robertf4) [reply _inc/client/setup-wizard/income-question/index.jsx]: Good catch, done!

File: _inc/client/setup-wizard/checklist-answer/index.jsx
Comment (robertf4) [reply _inc/client/setup-wizard/checklist-answer/index.jsx:15]: I started with this as a functional component but I believe I couldn't get one of the code checker tools to accept it because the `set` function from `useState()` is generally called from an arrow function which isn't accepted by one of the code checkers. So I elected to switch to a class component and just put the functions on the class to avoid arrow functions.

I believe I could instead declare functions like `function foo() { ... }` inside a functional component, but that doesn't avoid the root problem of creating a function on each render so I decided not to. For each call to `setState` I would have to create a corresponding function like `function setVariableState( prop ) { setVariable( prop ); }` to avoid having to use an arrow function like `onChange={ ( prop ) => { setVariable( prop ) }` in the JSX.

All that said, I do prefer functional components to class components so I'd like to see us solve this!

File: _inc/client/setup-wizard/intro-page/index.jsx
Comment (robertf4) [reply _inc/client/setup-wizard/intro-page/index.jsx:41]: I plan to add Tracks events across the entire setup wizard in one PR, after all the pages are merged. That way all the new events will be in one place and I can make a P2 to get design and marketing review on all the event names and props.

**Comment by robertf4** [comment]
> it seems like some padding is missing above each checkbox

It was, I've added padding now.

File: _inc/client/setup-wizard/checklist-answer/index.jsx
Comment (robertf4) [reply _inc/client/setup-wizard/checklist-answer/index.jsx]: We discussed this in Slack but I wanted to put it here for anyone who may be following:

This issue is solved by the `useCallback` hook as shown in the current functional implementation in this PR.

**Review by jeherve — DISMISSED** [review]
This looks good to me! If @jeffgolenski is okay with this, let's ship it.

**Review by jeherve — APPROVED** [review]

**Pull request merged**
